l10n_ec_withhold/wizard/wizard_create_sale_withhold.py

```diff
@@ -95,6 +118,47 @@ def validate_repeated_invoice(self):
                     )
                 )
 
+    def validate_repeated_withhold(self):
+        result = self.env["account.move"].search(
+            [
+                ("partner_id", "=", self.partner_id.id),
+                ("name", "=", f"RET {self.document_number}"),
+                ("l10n_ec_withholding_type", "=", "sale"),
+                ("l10n_latam_internal_type", "=", "withhold"),
+                ("l10n_ec_journal_type", "=", "sale"),
+            ]
+        )
+        if len(result) > 0:
+            raise UserError(_(f"Withhold {self.document_number} already exist"))
+
+    def validate_selected_invoices(self):
+        line_invoices = []
+        for withhold_line in self.withhold_line_ids:
+            line_invoices.append(withhold_line.invoice_id.name)
+        invoices = []
+        for invoice in self.invoice_ids:
+            invoices.append(invoice.name)
+
+        if set(line_invoices) != set(invoices):
+            raise UserError(_("Withhold not content selected invoices"))
+
+    def validate(self):
+        if not self.withhold_line_ids:
+            raise UserError(_("Please add some withholding lines before continue"))
+
+        withhold_date = self.issue_date.strftime("%Y-%m-%d")
+        for invoice in self.invoice_ids:
+            invoice_date = invoice.invoice_date.strftime("%Y-%m-%d")
+            if withhold_date < invoice_date:
```

Comment:
Veo que estas convirtiendo a string la fecha para compararlas, lo ideal es compararlas como fechas, es una mejor comparacion y no haces un cast del tipo de datos
```suggestion
            if self.issue_date < invoice.invoice_date:
```